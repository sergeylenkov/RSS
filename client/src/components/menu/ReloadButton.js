import React from 'react';
import { Icon, Icons } from '../Icon.js';

import styles from './ReloadButton.module.css';

export class ReloadButton extends React.Component {
    render() {
        let className = styles.container;

        if (this.props.isActive) {
            className += ` ${styles.active}`;
        }

        return (
            <button className={className} onClick={() => this.props.onClick()}>
                <div className={styles.icon}><Icon svg={Icons.reload}/></div>
                <div className={styles.label}>Свежее</div>
                <div className={styles.counter}>{this.props.count}</div>
            </button>
        );        
    }
}