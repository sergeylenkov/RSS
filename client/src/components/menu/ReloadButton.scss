.reload_button {
	&__container {
		display: flex;
		flex-direction: row;
		flex-shrink: 0;
		overflow: hidden;
		margin: 0;
		opacity: 1;
		transition: opacity 0.2s ease-in-out;

		&--active {
			opacity: 0.6;
			pointer-events: none;
		}
	}

	&__icon {
  	width: 16px;
		height: 16px;
		margin: auto 10px auto 0;
		stroke: #000000;
		stroke-width: 2px;
		fill: none;
		cursor: pointer;

		&--active {
			animation: spin 2s linear infinite;
		}

		&--error {
			stroke: #ff0000;
		}
	}

	&__icon > svg {
		width: 100%;
  	height: 100%;
	}
}

/*
* theme
*/

.dark .reload_button {
	&__icon {
		stroke: #f3f3f3;
	}
}

@keyframes spin {
	100% {
		transform: rotate(360deg);
	}
}
