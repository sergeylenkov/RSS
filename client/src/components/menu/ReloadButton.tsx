import MenuButton from './MenuButton';
import React from 'react';
import { ReloadIcon } from '../Icons';
import { Bem } from '../../utils/bem';

import './ReloadButton.scss';

const block = new Bem('reload-button');

interface ReloadButtonProps {
  isSelected: boolean;
  isActive: boolean;
  isError: boolean;
  count: number;
  onClick: () => void;
  onUpdate: () => void;
}

function ReloadButton({ isSelected, count, isActive, isError, onUpdate, onClick } : ReloadButtonProps): JSX.Element {
  const blockClass = block.addModifier(isActive ? 'active' : undefined).build();
  const buttonClass = block.getElement('icon').addModifier(isActive ? 'active' : undefined).addModifier(isError ? 'error' : undefined).build();

  return (
    <div className={blockClass}>
      <button className={buttonClass} onClick={onUpdate}><ReloadIcon /></button>
      <MenuButton title={'Свежее'} isSelected={isSelected} count={count} onClick={onClick} />
    </div>
  );
}

export default ReloadButton;
