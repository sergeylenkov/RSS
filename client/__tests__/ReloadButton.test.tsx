import React from 'react';
import renderer  from 'react-test-renderer';
import ReloadButton from '../src/components/menu/ReloadButton';

const action = () => { return };

test('ReloadButton correct rendering', () => {
  const render = renderer.create(<ReloadButton isActive isError={false} isSelected={false} count={10} onClick={action} onUpdate={action} />);
  expect(render).toMatchSnapshot();
});