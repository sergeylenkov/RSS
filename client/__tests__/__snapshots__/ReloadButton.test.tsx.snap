// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReloadButton correct rendering 1`] = `
<div
  className="reload-button reload-button_active"
>
  <button
    className="reload-button__icon reload-button__icon_active"
    onClick={[Function]}
  >
    <svg
      viewBox="0 0 24 24"
    >
      <g>
        <polyline
          points="23 4 23 10 17 10"
        />
        <polyline
          points="1 20 1 14 7 14"
        />
        <path
          d="M3.51 9a9 9 0 0 1 14.85-3.36L23 10M1 14l4.64 4.36A9 9 0 0 0 20.49 15"
        />
      </g>
    </svg>
  </button>
  <button
    className="menu-button"
    onClick={[Function]}
  >
    <div
      className="menu-button__label"
    >
      Свежее
    </div>
  </button>
</div>
`;
